bl/interfaces/controller.go

```diff
@@ -0,0 +1,47 @@
+package interfaces
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/kimbellG/kerror"
+	ttgrpc "github.com/kimbellG/tournament-bl/interfaces/grpc"
+	"github.com/kimbellG/tournament-bl/models"
+	"github.com/kimbellG/tournament-bl/usecases"
+)
+
+type ServiceController struct {
+	userUsecase usecases.UserUsecase
+}
+
+func (sc *ServiceController) SaveUser(ctx context.Context, user *ttgrpc.User) (*ttgrpc.SaveResponse, error) {
+	mUser, err := userGrpcToModels(user)
```

Comment:
The idea of converting is correct, but this is DB layer and it has nothing to do with grpc stuff. Just convert it to models in grpc layer.